strax/storage/rucio.py

```diff
@@ -0,0 +1,41 @@
+import json
+import hashlib
+import os.path as osp
+
+import strax
+from strax.storage.files import dirname_to_prefix
+
+export, __all__ = strax.exporter()
+
+
+@export
+class rucio(strax.StorageBackend):
+    """Get data from a rucio directory
+    """
+
+    def get_metadata(self, dirname, **kwargs):
+        dirname = str(dirname)
+        prefix = dirname_to_prefix(dirname)
+        metadata_json = f'{prefix}-metadata.json'
+        fn = rucio_path(metadata_json, dirname)
+        with open(fn, mode='r') as f:
+            return json.loads(f.read())
+
+    def _read_chunk(self, dirname, chunk_info, dtype, compressor):
+        #print('yes')
+        fn = rucio_path(chunk_info['filename'], dirname)
+        return strax.load_file(fn, dtype=dtype, compressor=compressor)
+
+    def _saver(self, dirname, metadata):
+        raise NotImplementedError(
+            "Cannot save directly into rucio, upload with admix instead")
+
+
+def rucio_path(filename, dirname):
+    root_path ='/dali/lgrandi/rucio'
```

Comment:
We shouldn't hard-code it like this as this might change to e.g. '/dali/lgrandi/xenonnt/rucio' I guess you want to get the info from the dirname.

Reply:
Unfortunately, dirname does not include the root path 😕

Reply:
Hmm okay. So how about this:

You pass it on at initialization:
https://github.com/XENONnT/straxen/blob/5b762f32eec7d3fb8a1a96e31ae504fe8c8cf3cd/straxen/rundb.py#L110
and set the default to /dali/lgrandi/rucio at the init of the rundb here:
https://github.com/XENONnT/straxen/blob/5b762f32eec7d3fb8a1a96e31ae504fe8c8cf3cd/straxen/rundb.py#L42

You would also have to change these lines:
https://github.com/AxFoundation/strax/blob/6404ae5f5733d1f53301bf0976fcc9f637acbfe3/strax/storage/rucio.py#L15 (with adding a def __init___ (..) ) and
https://github.com/AxFoundation/strax/blob/6404ae5f5733d1f53301bf0976fcc9f637acbfe3/strax/storage/rucio.py#L25

If you want some help I can also commit this idea to prevent hardcodes?